test/index.test.js

```diff
@@ -0,0 +1,1045 @@
+            getPrInfo: sinon.stub(),
+            stats: sinon.stub(),
+            canHandleWebhook: sinon.stub(),
+            getScmContexts: sinon.stub(),
+            getDisplayName: sinon.stub()
+        };
+        initMock(gitlabScmMock, 'gitlab');
+        exampleScmMock = {
+            dummyFunction: sinon.stub(),
+            dummyRejectFunction: sinon.stub(),
+            addWebhook: sinon.stub(),
+            parseUrl: sinon.stub(),
+            parseHook: sinon.stub(),
+            getCheckoutCommand: sinon.stub(),
+            decorateUrl: sinon.stub(),
+            decorateCommit: sinon.stub(),
+            decorateAuthor: sinon.stub(),
+            getPermissions: sinon.stub(),
+            getCommitSha: sinon.stub(),
+            updateCommitStatus: sinon.stub(),
+            getFile: sinon.stub(),
+            getOpenedPRs: sinon.stub(),
+            getBellConfiguration: sinon.stub(),
+            getPrInfo: sinon.stub(),
+            stats: sinon.stub(),
+            canHandleWebhook: sinon.stub(),
+            getScmContexts: sinon.stub(),
+            getDisplayName: sinon.stub()
+        };
+        initMock(exampleScmMock, 'example');
+
+        mockery.registerMock('screwdriver-scm-github', testScm(githubScmMock));
+        mockery.registerMock('screwdriver-scm-gitlab', testScm(gitlabScmMock));
+        mockery.registerMock('screwdriver-scm-example', testScm(exampleScmMock));
+
+        // eslint-disable-next-line global-require
+        Scm = require('../index');
+
+        scm = new Scm({
+            ecosystem,
+            scms: [
+                {
+                    plugin: 'github',
+                    config: githubPluginOptions
+                },
+                {
+                    plugin: 'example',
+                    config: examplePluginOptions
+                },
+                {
+                    plugin: 'gitlab',
+                    config: gitlabPluginOptions
+                }
+            ]
+        });
+    });
+
+    afterEach(() => {
+        mockery.deregisterAll();
+    });
+
+    describe('construction', () => {
+        let githubOptions;
+        let exampleOptions;
+        let gitlabOptions;
+
+        beforeEach(() => {
+            githubOptions = Object.assign(
+                { ecosystem },
+                githubPluginOptions
+            );
+            exampleOptions = Object.assign(
+                { ecosystem },
+                examplePluginOptions
+            );
+            gitlabOptions = Object.assign(
+                { ecosystem },
+                gitlabPluginOptions
+            );
+        });
+
+        it('throws an error when config does not exist', () => {
+            const error = new Error('No scm config passed in.');
+
+            try {
+                scm = new Scm();
+            } catch (err) {
+                assert.strictEqual(err.message, error.message);
+
+                return;
+            }
+            assert.fail();
```

Comment:
This can be written much more cleanly.

```
assert.throws(() => {
  scm = new Scm();
), Error, 'No scm config passed in.');
```

Try/catch in tests is unnecessary.